packages/flutter/lib/src/material/material_localizations.dart

```diff
@@ -434,6 +475,23 @@ class DefaultMaterialLocalizations implements MaterialLocalizations {
     'December',
   ];
 
+  /// Returns the number of days in a month, according to the proleptic
+  /// Gregorian calendar.
+  ///
+  /// This applies the leap year logic introduced by the Gregorian reforms of
+  /// 1582. It will not give valid results for dates prior to that time.
+  int _getDaysInMonth(int year, int month) {
```

Comment:
It's too bad this isn't already part of DateTime.  Maybe propose a PR for the Dart team?

Reply:
Yeah, I will put that on my list and follow up later. There are other things like the DateTimeRange and other stuff in the date_util.dart that might be good candidates as well.